apps/web/src/pages/Pools/ViewPoolPage/components/PublishProcessDialog.tsx

```diff
@@ -0,0 +1,147 @@
+import { commonMessages } from "@gc-digital-talent/i18n";
+
+import { ProcessDialogProps } from "./types";
+
+type PublishProcessDialogProps = ProcessDialogProps & {
+  closingDate: Pool["closingDate"];
+  onPublish: () => Promise<void>;
+};
+
+const PublishProcessDialog = ({
+  poolName,
+  closingDate,
+  isFetching,
+  onPublish,
+}: PublishProcessDialogProps) => {
+  const [isOpen, setIsOpen] = React.useState<boolean>(false);
+  const intl = useIntl();
+
+  const title = intl.formatMessage({
+    defaultMessage: "Publish advertisement",
+    id: "YyT0ua",
+    description: "Title to publish a process' advertisement",
+  });
+
+  let closingStringLocal;
+  let closingStringPacific;
+  if (closingDate) {
+    const closingDateObject = parseDateTimeUtc(closingDate);
+    closingStringLocal = relativeClosingDate({
+      closingDate: closingDateObject,
+      intl,
+    });
+    closingStringPacific = relativeClosingDate({
+      closingDate: closingDateObject,
+      intl,
+      timeZone: "Canada/Pacific",
+    });
+  }
+
+  const handlePublish = async () => {
+    await onPublish().then(() => {
+      setIsOpen(false);
+    });
+  };
+
+  return (
+    <Dialog.Root open={isOpen} onOpenChange={setIsOpen}>
+      <Dialog.Trigger>
+        <Button color="primary" mode="inline">
+          {title}
+        </Button>
+      </Dialog.Trigger>
+      <Dialog.Content>
+        <Dialog.Header>{title}</Dialog.Header>
+        <Dialog.Body>
+          <p data-h2-margin-bottom="base(x1)">
+            {intl.formatMessage(
+              {
+                id: "FDZh4D",
+                defaultMessage:
+                  "You are about to publish the advertisement for this process: {poolName}",
+                description: "Text to confirm the process to be published",
+              },
+              {
+                poolName,
+              },
+            )}
+          </p>
+          <p data-h2-margin="base(x1, 0)">
+            {intl.formatMessage({
+              defaultMessage:
+                "This will make your advertisement available to applicants to submit applications.",
+              id: "bok2jV",
+              description: "Second paragraph for publish process dialog",
+            })}
+          </p>
+          <p data-h2-margin="base(x1, 0)">
+            {intl.formatMessage({
+              defaultMessage: "This process is set to automatically close on:",
+              id: "o4EnIo",
+              description: "Third paragraph for publish process dialog",
+            })}
+          </p>
+          <ul>
+            <li>
+              <strong>
+                {intl.formatMessage({
+                  defaultMessage: "Closing Date",
+                  id: "K+roYh",
+                  description:
+                    "Closing Date field label for publish pool dialog",
+                })}
+                {intl.formatMessage(commonMessages.dividingColon)}
+              </strong>
+              {closingStringLocal}
```

Comment:
Hmm, if the draft pool doesn't have a closing date yet, then this shows up empty. We would need to talk to design, but maybe we can add a link to the edit pool page #closing-date so that it's not empty here.

![image](https://github.com/GCTC-NTGC/gc-digital-talent/assets/22059495/e73ef6a2-53b7-4e19-b4a7-6c435a490864)

Reply:
@Jerryescandon What do you think?

Reply:
That specific dialog is being removed in https://github.com/GCTC-NTGC/gc-digital-talent/issues/7915

Reply:
It's a good idea, but yeah, that dialog is being removed when cleanup happens. I wouldn't sweat it for now.